### PR TITLE
Adds new set of AWS codebundles

### PR DESCRIPTION
**Pull request by jon-funk**
- EKS fargate health scan
- Elasticache redis scan
- S3 report
- Lambda Health scan

### CODE REVIEW

File: codebundles/aws-lambda-health/.runwhen/generation-rules/aws-lambda-health.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          mode: substring
+      slxs:
+        - baseName: aws-lambda-health
+          qualifiers: ["resource"]
+          baseTemplateName: aws-lambda-health
+          levelOfDetail: basic
+          outputItems:
+            - type: slx
+            - type: sli
+              templateName: aws-lambda-health-sli.yaml
```
Comment (stewartshea) [inline codebundles/aws-lambda-health/.runwhen/generation-rules/aws-lambda-health.yaml:21]: don't need the templateName here

File: codebundles/aws-lambda-health/.runwhen/templates/aws-lambda-health-taskset.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: runwhen.com/v1
+kind: Runbook
+metadata:
+  name: {{slx_name}}
+  labels:
+    {% include "common-labels.yaml" %}
+  annotations:
+    {% include "common-annotations.yaml" %}
+spec:
+  locations:
```
Comment (stewartshea) [inline codebundles/aws-lambda-health/.runwhen/templates/aws-lambda-health-taskset.yaml:10]: location: {{default_location}}

File: codebundles/aws-s3-bucket-storage-report/.runwhen/templates/aws-s3-bucket-storage-report-taskset.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: runwhen.com/v1
+kind: Runbook
+metadata:
+  name: {{slx_name}}
+  labels:
+    {% include "common-labels.yaml" %}
+  annotations:
+    {% include "common-annotations.yaml" %}
+spec:
+  locations:
```
Comment (stewartshea) [inline codebundles/aws-s3-bucket-storage-report/.runwhen/templates/aws-s3-bucket-storage-report-taskset.yaml:10]: location: {{default_location}}

**Review by stewartshea — CHANGES_REQUESTED** [review]
Some minor notes, mostly to share some knowledge. I see other example where runbook has a list of locations, so maybe both ways work, but in general the runbook should be a singular value

File: codebundles/aws-eks-health/runbook.robot [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    [Documentation]   This script checks the health status of an Amazon EKS Fargate cluster.
+    [Tags]  EKS    Fargate    Cluster Health    AWS    Kubernetes    Pods    Nodes  
+    ${process}=    RW.CLI.Run Bash File    check_eks_fargate_cluster_health_status.sh
+    ...    env=${env}
+    ...    secret__AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+    ...    secret__AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+    IF    "Error" in """${process.stdout}"""
+        RW.Core.Add Issue    title=EKS Fargate Cluster in ${AWS_REGION} is Unhealthy
+        ...    severity=3
+        ...    next_steps=Check the AWS Management Console for the health status of the EKS Fargate cluster.
```
Comment (jon-funk) [inline codebundles/aws-eks-health/runbook.robot:29]: Update to "fetch X logs from cloudtrail"

File: codebundles/aws-eks-health/.runwhen/generation-rules/aws-eks-health.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: runwhen.com/v1
+kind: GenerationRules
+spec:
+  platform: aws
+  generationRules:
+    - resourceTypes:
```
Comment (jon-funk) [inline codebundles/aws-eks-health/.runwhen/generation-rules/aws-eks-health.yaml:6]: Send note for updating to "match on type and not instance"

File: codebundles/aws-elasticache-redis-health/runbook.robot [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    IF    "is not available" in """${process.stdout}"""
+        RW.Core.Add Issue    title=Elasticache instance(s) not available in region ${AWS_REGION}
+        ...    severity=2
+        ...    next_steps=Review metrics of the Elasticache instance(s) to determine the cause of the issue.
+        ...    expected=The Elasticache instance(s) should be available in the specified region.
+        ...    actual=The Elasticache instance(s) is not available in the specified region.
+        ...    reproduce_hint=Check the AWS Management Console for the status of the Elasticache instance(s).
+        ...    details=${process.stdout}
+    END
+
```
Comment (jon-funk) [inline codebundles/aws-elasticache-redis-health/runbook.robot:45]: add task for fetch cloudtrail logs

File: codebundles/aws-elasticache-redis-health/runbook.robot [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+# TODO: discuss vpc topology and ec2 bastion access for aws redis elasticache
+# Monitor Redis Performance using INFO and SLOWLOG commands
+#     [Documentation]   This script is used to monitor the performance of a Redis instance hosted on AWS Elasticache. It first retrieves the details of the Redis instance using the AWS CLI. Then, it connects to the Redis instance and uses the Redis INFO and SLOWLOG commands to monitor its performance. The output of these commands is printed to the console for review.
+#     [Tags]  AWS    ElastiCache    Redis    Performance Monitoring    SLOWLOG    INFO    BASH    CLI    Script    Cloud Services    Database    
+#     ${process}=    Run Process    ${CURDIR}/monitor_redis_performance.sh    env=${env}
+#     RW.Core.Add Pre To Report    ${process.stdout}
```
Comment (jon-funk) [inline codebundles/aws-elasticache-redis-health/runbook.robot:51]: cleanup

File: codebundles/aws-lambda-health/runbook.robot [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    ...    secret__AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+    ...    secret__AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+    RW.Core.Add Pre To Report    ${process.stdout}
+    IF    "ERROR" in """${process.stdout}"""
+        RW.Core.Add Issue    title=AWS Lambda Invocation Errors
+        ...    severity=2
+        ...    next_steps=Review Lambda logs for errors.
+        ...    expected=The Lambda function has no invocation errors.
+        ...    actual=The Lambda function has invocation errors.
+        ...    reproduce_hint=Check the AWS Management Console for the status of the Lambda function.
```
Comment (jon-funk) [inline codebundles/aws-lambda-health/runbook.robot:42]: update nextsteps (try generate link)

**Pull request merged**